appservice/src/createAppService/SiteRuntimeStep.ts

```diff
@@ -70,14 +70,16 @@ export function convertStacksToPicks(stacks: ApplicationStack[], recommendedRunt
         .map(stack => (stack.majorVersions || []).map(mv => {
             return {
                 runtimeVersion: nonNullProp(mv, 'runtimeVersion'),
-                displayVersion: nonNullProp(mv, 'displayVersion'),
-                stackDisplay: nonNullProp(stack, 'display')
+                stackDisplay: nonNullProp(stack, 'display'),
+                // include stack display in the display name only if it doesn't include a version number
+                // this simplifies names such as 'Java 8 Tomcat 8.5' to just 'Tomcat 8.5'
+                displayName: /[0-9]/.test(nonNullProp(stack, 'display')) ? nonNullProp(mv, 'displayVersion') : `${nonNullProp(stack, 'display')} ${nonNullProp(mv, 'displayVersion')}`
```

Comment:
Could just take out `nonNullProp(stack, 'display')` and make it a variable so you don't have to call it 3 separate times.